### ISSUE TITLE
Add an option to set relative build path

### ISSUE THREAD

**Issue by ivan-volnov** [cdelledonne/vim-cmake#19]
It should be configurable to set path for build folder. For example '..' (parent directory of the project root)

```
/Users/user/project/
├── Debug
    └── compile_commands.json
└── project
    ├── .git
    ├── CMakeLists.txt
    ├── compile_commands.json -> ../Debug/compile_commands.json
    └── src
        └── main.cpp
```

**Comment by cdelledonne** [comment]
Apologies for the delay on this, at the moment I am quite busy with some other stuff and cannot work on this plugin too much. I will take this up in a few weeks.

**Comment by cdelledonne** [comment]
I've added a new configuration option, `g:cmake_build_dir_location`, please run `:help cmake-configuration` to read about it.

I have also improved the usage of `:CMakeGenerate` as well, please run `:help cmake-generate` to check out the new usage.

Closed in 6318c72321d9be3ac9c1fc1dac88531069472d78
